src/course-unit/move-modal/utils.ts

```diff
@@ -0,0 +1,120 @@
+import { BASIC_BLOCK_TYPES, CATEGORIES_KEYS } from './constants';
+import { ITreeNode, IXBlockInfo, IAncestor } from './interfaces';
+import messages from './messages';
+
+/**
+ * Determines the XBlock type based on the provided category and parent information.
+ *
+ * @param {string} category - The category of the XBlock (e.g., 'chapter', 'sequential', 'vertical').
+ * @returns {string} - The determined XBlock type (e.g., 'section', 'subsection', 'unit').
+ */
+export const getXBlockType = (category: string): string => {
+  switch (true) {
+    case category === CATEGORIES_KEYS.chapter:
+      return CATEGORIES_KEYS.section;
+    case category === CATEGORIES_KEYS.sequential:
+      return CATEGORIES_KEYS.subsection;
+    case category === CATEGORIES_KEYS.vertical:
+      return CATEGORIES_KEYS.unit;
+    default:
+      return category;
+  }
+};
```

Comment:
[optional]: I consider `switch(true)` an anti-pattern if you also suggest using the concatenated approach :)

```suggestion
export const getXBlockType = (category: string): string => {
  const categoryMap: { [key: string]: string } = {
    [CATEGORIES_KEYS.chapter]: CATEGORIES_KEYS.section,
    [CATEGORIES_KEYS.sequential]: CATEGORIES_KEYS.subsection,
    [CATEGORIES_KEYS.vertical]: CATEGORIES_KEYS.unit,
  };

  return categoryMap[category] || category;
};
```

Reply:
Changed